deepchecks/checks/performance/performance_report.py

```diff
@@ -45,41 +70,186 @@ def run(self, dataset, model=None) -> CheckResult:
         Returns:
             CheckResult: value is dictionary in format 'score-name': score-value
         """
-        return self._performance_report(dataset, model)
+        return self._performance_report(train_dataset, test_dataset, model)
 
-    def _performance_report(self, dataset: Dataset, model):
-        Dataset.validate_dataset(dataset)
-        dataset.validate_label()
-        validate_model(dataset, model)
+    def _performance_report(self, train_dataset: Dataset, test_dataset: Dataset, model):
+        Dataset.validate_dataset(train_dataset)
+        Dataset.validate_dataset(test_dataset)
+        train_dataset.validate_label()
+        test_dataset.validate_label()
+        train_dataset.validate_shared_label(test_dataset)
+        train_dataset.validate_shared_features(test_dataset)
+        validate_model(test_dataset, model)
+
+        task_type = task_type_check(model, train_dataset)
 
         # Get default scorers if no alternative, or validate alternatives
-        scorers = get_scorers_list(model, dataset, self.alternative_scorers)
-        scores = {
-            scorer.name: scorer(model, dataset)
-            for scorer in scorers
-        }
+        scorers = get_scorers_list(model, test_dataset, self.alternative_scorers, multiclass_avg=False)
+        datasets = {'Train': train_dataset, 'Test': test_dataset}
+        if task_type in [ModelType.BINARY, ModelType.MULTICLASS]:
+            x = ['Class', 'Dataset']
+            results = []
+            for dataset_name, dataset in datasets.items():
+                for scorer in scorers:
+                    score_result = scorer(model, dataset)
+                    # Multiclass scorers return numpy array of result per class
+                    for class_i, value in enumerate(score_result):
```

Comment:
what about binary scorers?

Reply:
This won't work if the labels are not consecutive (for example the labels are 1,2,6)
use this: 
```
for class_score, class_value in zip(score_result, test_dataset.classes):
```